airbyte-integrations/connectors/source-gitlab/source_gitlab/streams.py

```diff
@@ -0,0 +1,334 @@
+        return f"groups/{stream_slice['id']}"
+
+    def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
+        for gid in self.group_ids:
+            yield {"id": gid}
+
+    def transform(self, record, stream_slice: Mapping[str, Any] = None, **kwargs):
+        record["projects"] = [
+            {"id": project["id"], "path_with_namespace": project["path_with_namespace"]} for project in record.pop("projects", [])
+        ]
+        return record
+
+
+class Projects(GitlabStream):
+    stream_base_params = {"statistics": 1}
+
+    def __init__(self, project_ids: List = None, parent_stream: GitlabStream = None, **kwargs):
+        super().__init__(**kwargs)
+        self.project_ids = project_ids
+        self.parent_stream = parent_stream
+
+    def path(self, stream_slice: Mapping[str, Any] = None, **kwargs) -> str:
+        return f"projects/{stream_slice['id']}"
+
+    def stream_slices(self, **kwargs) -> Iterable[Optional[Mapping[str, any]]]:
+        if self.parent_stream:
+            group_project_ids = set()
+            for slice in self.parent_stream.stream_slices(sync_mode=SyncMode.full_refresh):
+                for record in self.parent_stream.read_records(sync_mode=SyncMode.full_refresh, stream_slice=slice):
+                    group_project_ids.update({i["path_with_namespace"] for i in record["projects"]})
+                for pid in group_project_ids:
+                    if not self.project_ids or self.project_ids and pid in self.project_ids:
+                        yield {"id": pid.replace("/", "%2F")}
+        else:
+            for pid in self.project_ids:
+                yield {"id": pid.replace("/", "%2F")}
+
+
+class Milestones(GitlabChildStream):
+    pass
+
+
+class Members(GitlabChildStream):
+    flatten_parent_id = True
+
+
+class Labels(GitlabChildStream):
+    flatten_parent_id = True
+
+
+class Branches(GitlabChildStream):
+    primary_key = "name"
+    flatten_id_keys = ["commit"]
+    flatten_parent_id = True
+
+
+class Commits(IncrementalGitlabChildStream):
+    cursor_field = "created_at"
+    filter_field = "since"
+    flatten_parent_id = True
+    stream_base_params = {"with_stats": True}
+
+
+class Issues(IncrementalGitlabChildStream):
+    stream_base_params = {"scope": "all"}
+    flatten_id_keys = ["author", "assignee", "closed_by", "milestone"]
+    flatten_list_keys = ["assignees"]
+
+
+class MergeRequests(IncrementalGitlabChildStream):
+    stream_base_params = {"scope": "all"}
+    flatten_id_keys = ["author", "assignee", "closed_by", "milestone", "merged_by"]
+    flatten_list_keys = ["assignees"]
+
+
+class MergeRequestCommits(GitlabChildStream):
+    path_list = ["project_id", "iid"]
+
+    path_template = "projects/{project_id}/merge_requests/{iid}"
+
+    def transform(self, record, stream_slice: Mapping[str, Any] = None, **kwargs):
+        record["project_id"] = stream_slice["project_id"]
```

Comment:
not sure if we skipped call to super intentionally here

Reply:
Here I have overridden the method, since we only need two additional parent fields.
`super` is not needs here.

Reply:
the problem that I see is that we break the chain of inheritance and super() calls, without any strong reason (having supper() call and merge of result will give the same effect, right?).
This means child classes knows details of the implementation of their parent and parent has no control now over general logic:
- lets say we want to fix the format of `primary_key` column
- before it was possible to fix this on the parent level without modifying child classes
- now we have to implement this logic in multiple places

child classes should extend parent logic and override it only if parent logic is not applicable or destructible for child